code/modules/reagents/chemistry/recipes/drugs.dm

```diff
@@ -14,10 +14,71 @@
 	mix_message = "The mixture dries into a pale blue powder."
 	required_temp = 380
 
+
 /datum/chemical_reaction/methamphetamine
 	results = list(/datum/reagent/drug/methamphetamine = 4)
 	required_reagents = list(/datum/reagent/medicine/ephedrine = 1, /datum/reagent/iodine = 1, /datum/reagent/phosphorus = 1, /datum/reagent/hydrogen = 1)
-	required_temp = 374
+	required_temp = 372
+	optimal_temp = 376//Wow this is tight
+	overheat_temp = 380 
+	optimal_ph_min = 6.5
+	optimal_ph_max = 7.5
+	determin_ph_range = 5
+	temp_exponent_factor = 1
+	ph_exponent_factor = 1.4
+	thermic_constant = 0.1 //exothermic nature is equal to impurty
+	H_ion_release = -0.025 
+	rate_up_lim = 10
+	purity_min = 0.5 //100u will natrually just dip under this w/ no buffer
+	reaction_flags = REACTION_HEAT_ARBITARY //Heating up is arbitary because of submechanics of this reaction.
+	
+//The less pure it is, the faster it heats up. tg please don't hate me for making your meth even more dangerous
+/datum/chemical_reaction/methamphetamine/reaction_step(datum/equilibrium/reaction, datum/reagents/holder, delta_t, delta_ph, step_reaction_vol)
+	var/datum/reagent/meth = holder.get_reagent(/datum/reagent/drug/methamphetamine)
+	if(!meth)//First step
+		reaction.thermic_mod = (1-delta_ph)*5
+		return
+	reaction.thermic_mod = (1-meth.purity)*5
+
+/datum/chemical_reaction/methamphetamine/overheated(datum/reagents/holder, datum/equilibrium/equilibrium)
+	. = ..()
+	temp_meth_explosion(holder, equilibrium.reacted_vol)
+
+/datum/chemical_reaction/methamphetamine/overly_impure(datum/reagents/holder, datum/equilibrium/equilibrium)
+	. = ..()
+	temp_meth_explosion(holder, equilibrium.reacted_vol)
+
+/datum/chemical_reaction/methamphetamine/reaction_finish(datum/reagents/holder, react_vol)
+	var/datum/reagent/meth = holder.get_reagent(/datum/reagent/drug/methamphetamine)
+	if(!meth)//Other procs before this can already blow us up
+		return ..()
+	if(meth.purity < purity_min)
+		temp_meth_explosion(holder, react_vol)
+	else
+		return ..()
```

Comment:
```suggestion
/datum/chemical_reaction/methamphetamine/reaction_finish(datum/reagents/holder, react_vol)
	var/datum/reagent/meth = holder.get_reagent(/datum/reagent/drug/methamphetamine)
	if(!meth)//Other procs before this can already blow us up
		return ..()
	if(meth.purity < purity_min)
		temp_meth_explosion(holder, react_vol)
		return
	return ..()
```